src/main/java/uk/org/okapibarcode/backend/Pdf417AutoStructuredAppend.java

```diff
@@ -0,0 +1,153 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.org.okapibarcode.backend;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+
+/**
+ * @author Urs Wolfer
+ */
+public class Pdf417AutoStructuredAppend {
+
+    /**
+     * Splits up the data into a series of structured append PDF417 symbols which include
+     * segment macro data <code>total</code> and <code>position</code>.
+     * Input data will be assumed to be of the type set by {@link Pdf417#setDataType(Symbol.DataType)}.
+     *
+     * @param data     the data to encode
+     * @param template the PDF417 symbol template which will be used for all created
+     *                 symbols. The following properties will be ignored from the template:
+     *                 <ul>
+     *                      <li>{@link Pdf417#setContent(String)}</li>
+     *                      <li>{@link Pdf417#setContent(byte[])}</li>
+     *                      <li>{@link Pdf417#setStructuredAppendTotal(int)}</li>
+     *                      <li>{@link Pdf417#setStructuredAppendPosition(int)}</li>
+     *                 </ul>
+     * @throws OkapiException if no data or data is invalid
+     */
+    public static List<Pdf417> createStructuredAppendSymbols(String data, Pdf417 template) {
+        List<String> dataList = splitData(data, template);
+        return plotStructuredAppendSymbols(dataList, template);
+    }
+
+    /**
+     * Overloaded method of {@link #createStructuredAppendSymbols(String, Pdf417)} which
+     * accepts binary data and passes it to {@link Pdf417#setContent(byte[])}.
+     *
+     * <p><b>NOTE:</b>See linked method above for details.
+     */
+    public static List<Pdf417> createStructuredAppendSymbols(byte[] data, Pdf417 template) {
+        return createStructuredAppendSymbols(new String(data, ISO_8859_1), template);
+    }
+
+    private static List<String> splitData(String data, Pdf417 template) {
+        Pdf417 testSymbol = new Pdf417() {
+            @Override
+            protected void plotSymbol() {
+            } // expensive plotting is not required
+        };
+        clone(template, testSymbol);
+        testSymbol.setStructuredAppendTotal(2);
+
+        List<String> dataList = new ArrayList<>();
+        String remainingData = data;
+        while (!remainingData.isEmpty()) {
+            int low = 0;
+            int high = remainingData.length();
```

Comment:
@gredler Do you think we can limit the `high`? I.e. is there a way to calculate the maximum ever possible data amount for a specified symbol? That would allow us to optimize performance.